j5/boards/base.py

```diff
@@ -114,21 +113,16 @@ def __next__(self) -> Board:
         """Get the next item in the iteration."""
         if self._iterator_counter >= len(self.boards):
             raise StopIteration
-        board = self.boards[self._iterator_counter]
+        board = self.boards[list(self.boards.keys())[self._iterator_counter]]
```

Comment:
Please can we use `self.boards.values()`

```suggestion
        board = list(self.boards.values())[self._iterator_counter]
```